cmd/ocsp-responder/main.go

```diff
@@ -190,25 +210,163 @@ func (src *dbSource) Response(ctx context.Context, req *ocsp.Request) ([]byte, h
+		err := fmt.Errorf("looking up OCSP response for serial: %s err: %w", serialString, ctx.Err())
+		src.log.Debugf(err.Error())
+		ocspLookups.WithLabelValues("canceled").Inc()
+		return nil, nil, err
+	case primaryResult := <-primaryChan:
+		if primaryResult.err != nil {
+			src.log.AuditErrf("Looking up OCSP response: %s", err)
+			ocspLookups.WithLabelValues("mysql_failed").Inc()
+			sourceUsed.WithLabelValues("error_returned").Inc()
+			return nil, nil, primaryResult.err
+		}
+		// Parse the OCSP bytes returned from the primary source to check
+		// status, expiration and other fields.
+		primaryParsed, err := ocsp.ParseResponse(primaryResult.bytes, nil)
+		if err != nil {
+			src.log.AuditErrf("parsing OCSP response: %s", err)
+			ocspLookups.WithLabelValues("mysql_failed").Inc()
+			sourceUsed.WithLabelValues("error_returned").Inc()
+			return nil, nil, err
+		}
+		src.log.Debugf("returning ocsp from primary source: %v", helper.PrettyResponse(primaryParsed))
+		ocspLookups.WithLabelValues("mysql_success").Inc()
+		sourceUsed.WithLabelValues("mysql").Inc()
+		return primaryResult.bytes, header, nil
+	case secondaryResult := <-secondaryChan:
+		// If secondary returns first, wait for primary to return for
+		// comparison.
+		var primaryResult lookupResponse
+		// Listen for cancellation or timeout waiting for primary result.
+		select {
+		case <-ctx.Done():
+			err := fmt.Errorf("looking up OCSP response for serial: %s err: %w", serialString, ctx.Err())
+			src.log.Debugf(err.Error())
+			ocspLookups.WithLabelValues("canceled").Inc()
+			return nil, nil, err
+		case primaryResult = <-primaryChan:
+		}
+
+		if primaryResult.err != nil {
+			src.log.AuditErrf("Looking up OCSP response: %s", err)
+			ocspLookups.WithLabelValues("mysql_failed").Inc()
+			sourceUsed.WithLabelValues("error_returned").Inc()
+			return nil, nil, primaryResult.err
+		}
+		// Parse the OCSP bytes returned from the primary source to check
+		// status, expiration and other fields.
+		primaryParsed, err := ocsp.ParseResponse(primaryResult.bytes, nil)
+		if err != nil {
+			src.log.AuditErrf("parsing OCSP response: %s", err)
+			ocspLookups.WithLabelValues("mysql_failed").Inc()
+			sourceUsed.WithLabelValues("error_returned").Inc()
+			return nil, nil, err
+		}
+
+		secondaryParsed, err := ocsp.ParseResponse(secondaryResult.bytes, nil)
+		if err != nil {
+			src.log.Debugf("secondary OCSP lookup response error: %v", err)
+			ocspLookups.WithLabelValues("redis_failed").Inc()
+			sourceUsed.WithLabelValues("mysql").Inc()
+			return primaryResult.bytes, header, nil
+		}
+		if primaryParsed.Status != secondaryParsed.Status {
+			src.log.Err("primary ocsp source doesn't match secondary source, returning primary response")
+			ocspLookups.WithLabelValues("redis_mismatch").Inc()
+			sourceUsed.WithLabelValues("mysql").Inc()
+			return primaryResult.bytes, header, nil
+		}
+		src.log.Debugf("returning ocsp from secondary source: %v", helper.PrettyResponse(secondaryParsed))
```

Comment:
Let's remove this. It's a fairly big log entry. And I'd have to check but we might incur the CPU cost for formatting the string even if the log level is not > debug.

Reply:
Removed.